tasks/main.yml

```diff
@@ -1,4 +1,10 @@
 ---
+- name: Verify Ansible is modern enough to support `allow_downgrade` for yum+apt.
+  assert:
+    that: "ansible_version.full is version_compare('2.12', '>=')"
```

Comment:
The difficulty here is there are a _lot_ of users still running 2.9, and probably a good number running 2.10 and 2.11 in between, who would immediately be blocked from using this role if we go this route...

Reply:
Ah right, I kind of expected this. So, I believe the best thing is to put this patch back into draft mode?

If you see any other chance, to conditionally implement that feature for users already on `ansible-core>=2.12`, instead of croaking on the assertion, please let me know. As I am a newcomer to Ansible, I would need some guidance. On the other hand, I believe it would probably complicate things too much.

Reply:
P.S.: I also updated the patch to adjust `galaxy_info.min_ansible_version: 2.12` within `meta/main.yml` instead.

Reply:
Hi again,

based on your feedback, and learning some more bits of Ansible, I took another approach and updated this patch to use conditional dispatching between two tasks, responsible for installing the package, based on the version of `ansible-core`. In order to move on, I believe many roles and collections on Galaxy will start using that approach.

Earlier, I said:

> I believe [such a patch] would probably complicate things too much.

When looking at the patch now, I believe the impact is reasonable and solves an important detail for us and others, being able to roll back to previous package versions, even on `apt`-based systems.

With kind regards,
Andreas.